src/mds/CInode.cc

```diff
@@ -1909,6 +1914,8 @@ void CInode::decode_lock_ifile(bufferlist::const_iterator& p)
       }
     }
   }
+  if (is_auth() && struct_v >= 2)
+    decode(_inode->fscrypt_file, p);
```

Comment:
Here, it seems you should switch to:

`if (!is_auth() && struct_v >= 2)`

I think for mutiple MDS case, for the CInode only one MDS is the auth.

Reply:
I can't do that because (as @joscollin says below), _inode is not set unless is_auth() is set.

Still though, I don't see why any of this would lead to a crash. The encode and decode routines seem to be the reverse of one another. I'm wondering if maybe I need to extend some size value that I've missed, and that's causing the MDS to walk off the end of the buffer or something?

Reply:
It seems when it is `!is_auth()` will the `_inode` won't be NULL.
Please see https://github.com/ceph/ceph/pull/41284/commits/bd80964c6c0e2e8d00bc720b19e0abb0bc8aab6e#diff-70559ea96fce92c8b31e29d33acd5c752afd0976b8fce640d172a21d85ecf59aR1843.

Reply:
So you should switch to `if (!is_auth() && struct_v >= 2)` IMO.

Reply:
In none auth MDS, the `_inode` is not exist, so just allocate a dummy one.